pipeline_dp/pipeline_operations.py

```diff
@@ -77,6 +81,48 @@ def group_by_key(self, col, stage_name: str):
     def filter(self, col, fn, stage_name: str):
         return col | stage_name >> beam.Filter(fn)
 
+    def filter_partitions(self, col, public_partitions, data_extractors, stage_name: str):
+        """Filter out the partitions that are not not meant to be public.
+
+        Args:
+          col: input collection
+          public_partitions: collection of public partition keys
+          stage_name: name of the stage
+
+        Returns:
+          A filtered collection containing only data belonging to public_partitions
+
+        """
+        class PartitionsFilterJoin(beam.DoFn):
+          def process(self, joined_data):
+            key, rest = joined_data
+            values, is_public = rest.get('values'), rest.get('is_public')
+
+            # TODO the Issue #4 says this is blocked on other tasks. Revisit
+            # this once unblocked
+            if not values:
+              values = [None]
+
+            if is_public:
+              for value in values:
+                yield key, value
+
+        def is_public(col):
+          return col[0] in public_partitions
+
+        # no filtering if no public partitions are specified
+        if not public_partitions:
+          return col
+
+        col = col | beam.Map(lambda x: (data_extractors.partition_extractor(x), x))
+
+        pp_type = type(public_partitions)
+        if pp_type is list or pp_type is set:
+          return col | beam.Filter(is_public)
+        else:
```

Comment:
Great for implementing both in-memory and join based partition selection!

Reply:
Thanks!